analyses/molecular-subtyping-HGG/01-HGG-molecular-subtyping-defining-lesions.Rmd

```diff
@@ -55,35 +61,62 @@ if (!dir.exists(results_dir)) {
 metadata <-
   readr::read_tsv(file.path(root_dir, "data", "histologies-base.tsv"), 
                   guess_max = 100000) %>%
-  dplyr::filter(sample_type == "Tumor") %>%
-  dplyr::filter(cohort == "PBTA")
+  dplyr::filter(cohort %in% c("PBTA", "DGD"),
+                !is.na(pathology_diagnosis))
 
 # File from 00-LGAT-select-pathology-dx that is used for inclusion/exclusion
 path_dx_list_lgat <- jsonlite::fromJSON(
   file.path(lgat_subset_dir, 
             "lgat_subtyping_path_dx_strings.json")
 )
+
+
+# Read in the JSON file that contains the strings we'll use to include or
+# exclude samples for subtyping - see 00-HGG-select-pathology-dx
+path_dx_list <- jsonlite::fromJSON(
+  file.path(hgg_subset_dir,
+            "hgg_subtyping_path_dx_strings.json")
+)
+
+#### Filter metadata -----------------------------------------------------------
+
+# Filter metadata based on pathology diagnosis fields and include samples that
+# should be classified as high-grade glioma based on defining lesions
+
+# Samples included on the basis of the pathology diagnosis fields
+hgg_only_meta <- metadata %>%
+  # Inclusion on the basis of CBTTC harmonized pathology diagnoses
+  filter(pathology_diagnosis %in% path_dx_list$exact_path_dx |
+         # Inclusion based on pathology free text diagnosis
+         pathology_free_text_diagnosis == path_dx_list$gliomatosis_path_free_text_exact |
+         # Inclusion based on pathology free text diagnosis for IHG
+         pathology_free_text_diagnosis  %in% path_dx_list$IHG_path_free_path_dx)
+
+dna_bsids <- hgg_meta %>%
```

Comment:
`hgg_meta` is not created in previous code, Do you mean `hgg_only_meta`?

Reply:
good catch!